ls1intum/Artemis
# `Programming exercises`: Fix repository link for submissions in practice mode

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


For submissions in practice mode, clicking on git URL will lead to a 'page not found' error, because git links in practice mode need the string 'practice-' between exerciseId and userId (see screenshots below)

### Description

In order to fix this issue I added an if condition in ```programming-exercise.utils.ts``` to add the required string if the participation is a practice participation (boolean testRun).

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise where due date has already passed (student is still able to 'practice')

1. Log in to Artemis as student
2. Start the exercise in practice mode
3. Submit anything
4. Log in to Artemis as Instructor
5. Navigate to Course-Management and click the 'Participations' button on your exercise in practice mode
6. Click the participation id of an entry where practice is 'Yes'
7. Click the commit hash link

After the bugfix, this link should lead to the correct gitlab repository. 

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage







### Screenshots



Location of the bug
![Screenshot from 2023-08-18 10-02-19](https://github.com/ls1intum/Artemis/assets/105664200/537de71f-1268-431e-b678-bfb08db3a1c1)


Changes
![Screenshot from 2023-08-18 10-23-01](https://github.com/ls1intum/Artemis/assets/105664200/cb9d228c-b5a2-4fe0-9a29-edb4434420e0)
